### PR TITLE
PIM-4186: rework drop zone system

### PR DESCRIPTION
**Pull request by sumbobyboys**

### CODE REVIEW

**Comment by jmleroux** [comment]
Could be the good timing to start to comment methods ;)

**Comment by jmleroux** [comment]
Nice PR, the dropzone system is very clear

File: features/Context/Page/Base/Form.php [hunk 1]
```diff
@@ -620,10 +620,21 @@ protected function fillSelect2Field(NodeElement $label, $value)
...
                 $this->getSession()->wait(5000, '!$.active');
 
-                // Select the value in the displayed dropdown
-                if (null !== $item = $this->find('css', sprintf('#select2-drop li:contains("%s")', $value))) {
-                    $item->click();
-                }
+                $field = $this->spin(function () use ($value) {
+                    $item = $this->find('css', sprintf('#select2-drop li:contains("%s")', $value));
+
+                    if (null === $item) {
```
Comment (phaseinducer) [inline features/Context/Page/Base/Form.php]: `return $item ?: false;` is equivalent.

File: features/Context/Page/Base/Form.php [hunk 2]
```diff
@@ -620,10 +620,21 @@ protected function fillSelect2Field(NodeElement $label, $value)
...
+                        return false;
+                    } else {
+                        return $item;
+                    }
+
+                    echo "retry find file input" . PHP_EOL;
```
Comment (phaseinducer) [inline features/Context/Page/Base/Form.php]: Never reached ?

**Pull request merged**